### PR TITLE
FIX: Get only contact person lists of logged in user FIX: Update only contact person of current user FIX: Delete only contact person of current user OPT: Use HTTPResponseRedirect for going back to same page on views

### PR DESCRIPTION
**Pull request by ilhanmercan**

### CODE REVIEW

File: src/front/views.py [hunk 1]
```diff
@@ -248,10 +248,13 @@ class AccountContactUpdateView(UpdateView, LoginRequiredMixin):
     success_message = 'Contact person details successfully updated.'
     success_url = reverse_lazy('account_contacts')
 
+    def get_object(self, queryset=None):
+        return shortcuts.get_object_or_404(Contact, pk=self.kwargs.get('contact_id'), owner=self.request.user)
+
     def form_valid(self, form):
         if not zmaster.contact_create_update(form.instance):
             messages.error(self.request, self.error_message)
-            return redirect('account_contact_edit')
```
Comment (ilhanmercan) [inline src/front/views.py]: this was already broken before class based views. 
it was not returning the same page as there is also contact_id for the edit page. 
we could do 
`redirect('account_contact_edit', kwargs={'contact_id': self.kwargs.get('contact_id')})` 
but I think it wasn't that nice. 

So I changed this one and also other views with `HttpResponseRedirect(self.request.path_info)` which redirects you to same page that you are coming from.

File: src/front/views.py [hunk 2]
```diff
@@ -264,16 +267,22 @@ class AccountContactDeleteView(DeleteView, LoginRequiredMixin):
     success_message = 'Contact person successfully deleted.'
     success_url = reverse_lazy('account_contacts')
 
+    def get_object(self, queryset=None):
+        return shortcuts.get_object_or_404(Contact, pk=self.kwargs.get('contact_id'), owner=self.request.user)
```
Comment (ilhanmercan) [inline src/front/views.py]: this was forgotten from function based views.
user could delete any contact with changing the id in the URL

File: src/front/views.py [hunk 3]
```diff
@@ -264,16 +267,22 @@ class AccountContactDeleteView(DeleteView, LoginRequiredMixin):
...
 
 
-class AccountContactsView(ListView, LoginRequiredMixin):
+class AccountContactsListView(ListView, LoginRequiredMixin):
     template_name = 'front/account_contacts.html'
     model = Contact
     paginate_by = 10
 
+    def get_queryset(self):
+        return zcontacts.list_contacts(self.request.user)
```
Comment (ilhanmercan) [inline src/front/views.py]: this was forgotten from function based views.
user was seeing all contacts in the database. now it's filtered as it was before

File: src/front/views.py [hunk 4]
```diff
@@ -248,10 +248,13 @@ class AccountContactUpdateView(UpdateView, LoginRequiredMixin):
     success_message = 'Contact person details successfully updated.'
     success_url = reverse_lazy('account_contacts')
 
+    def get_object(self, queryset=None):
+        return shortcuts.get_object_or_404(Contact, pk=self.kwargs.get('contact_id'), owner=self.request.user)
```
Comment (ilhanmercan) [inline src/front/views.py]: this was forgotten from function based views.
user could update any contact with changing the id in the URL

File: src/front/views.py (hunk 3 above)
Comment (vesellov) [reply src/front/views.py]: haha :)

File: src/front/views.py (hunk 1 above)
Comment (vesellov) [reply src/front/views.py]: nice... also works

**Pull request merged**